enkit/outputs/commands.go

```diff
@@ -167,29 +159,19 @@ func (c *Mount) Run(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return fmt.Errorf("failed generating new buildbuddy client: %w", err)
 	}
-	bbOpts := bbexec.NewClientOptions(
-		&logger.DefaultLogger{Printer: log.Printf}, // TODO: pipe this logger everywhere
-		host,
-		port,
-		c.root.OutputsRoot,
-	)
-	_, err = bbexec.MaybeStartClient(bbOpts, 5*time.Second)
-	if err != nil {
-		return fmt.Errorf("failed to start bb_clientd: %w", err)
-	}
 	r, err := kbuildbarn.GenerateHardlinks(
 		context.Background(),
 		bc,
-		bbOpts.MountDir,
+                DefaultMountDir,
 		c.InvocationID,
-		host,
 		kbuildbarn.WithNamedSetOfFiles(),
 		kbuildbarn.WithTestResults(),
 	)
 	if err != nil {
 		return fmt.Errorf("hard links could not be generated: %w", err)
 	}
-	scratchInvocationPath := filepath.Join(bbOpts.ScratchDir(), c.InvocationID)
+
+	scratchInvocationPath := filepath.Join(DefaultOutputsRoot, c.InvocationID)
```

Comment:
outputsInvocationPath and scratchInvocationPath are the same path. Isn't that going to make the symlink below fail (or produce something weird?)

Reply:
Good catch!

In the original code we created hard links between the `cas` mount and a seperate `scratch` directory (the scratch directory is writable unlike the `cas` mount) (both these are fuse filesystems managed by buildbarn)

Then it created another symlink between the actual scratch, and the final output location we expose to the user:
- scratchInvocationPath: `/home/isaac/outputs/.bb_clientd/mount/scratch/<invocation-id>`
- outputInvocationPath: `/home/isaac/outputs/<invocation-id>`

When I refactored I didn't catch that I had made this the same path - and none of my testing surfaced any errors or weirdness.

We can just remove this symlink and `outputInvocationPath` altogether. It isn't necessary now that `enkit` is not responsible for setting up the mounts and needs some way to hide the `bb_clientd`  mount implementation details and present the user with a non-hidden path.

Now we already have an existing `/enf_mounts/buildbarn/scratch` mount we can point the user to.